cdap-ui/app/features/admin/controllers/namespace/templates-list-ctrl.js

```diff
@@ -1,27 +1,53 @@
 angular.module(PKG.name + '.feature.admin')
-  .controller('NamespaceTemplatesListController', function ($scope, mySettings, $stateParams, $alert) {
+  .controller('NamespaceTemplatesListController', function ($scope, mySettings, $stateParams, $alert, myHelpers) {
 
     var vm = this;
-    vm.list = {};
+    vm.list = [];
+
+    function objectToArray(obj) {
+      angular.forEach(obj, function (value) {
+        vm.list.push(value);
+      });
+    }
+
+    function processResult(response) {
+      if (response) {
+        vm.list = [];
+
+        var sources, transforms, sinks;
+
+        sources = myHelpers.objectQuery(response, $stateParams.nsadmin, 'ETLBatch', 'source');
+        transforms = myHelpers.objectQuery(response, $stateParams.nsadmin, 'ETLBatch', 'transform');
+        sinks = myHelpers.objectQuery(response, $stateParams.nsadmin, 'ETLBatch', 'sink');
+
+        objectToArray(sources);
+        objectToArray(transforms);
+        objectToArray(sinks);
+
+        sources = myHelpers.objectQuery(response, $stateParams.nsadmin, 'ETLRealtime', 'source');
+        transforms = myHelpers.objectQuery(response, $stateParams.nsadmin, 'ETLRealtime', 'transform');
+        sinks = myHelpers.objectQuery(response, $stateParams.nsadmin, 'ETLRealtime', 'sink');
+
+        objectToArray(sources);
+        objectToArray(transforms);
+        objectToArray(sinks);
+      }
+    }
 
 
     mySettings.get('pluginTemplates')
-      .then(function (res) {
-        if (res) {
-          vm.list = res[$stateParams.nsadmin];
-        }
-      });
+      .then(processResult);
 
     vm.isEmpty = function () {
-      return Object.keys(vm.list).length === 0;
+      return vm.list.length === 0;
     };
 
     vm.delete = function (template) {
       mySettings.get('pluginTemplates')
         .then(function (res) {
-          delete res[$stateParams.nsadmin][template.templateName];
+          delete res[$stateParams.nsadmin][template.templateType][template.type][template.templateName];
```

Comment:
This might be a little subjective, but ```template.templateType``` and ```template.type``` seems confusing. When I look at it at an instant its getting hard to settle down which one means which thing. There ```ETLBatch``` and ```ETLRealtime``` and then there is ```source, sink and transforms```. One is template type and the other is plugin type. I feel we should try changing it so that it becomes appropriate for us when we come back and see this after 2-3 months :smile: